### ISSUE TITLE
[Input] optionally display hint text below label

### ISSUE THREAD

**Issue by theetrain** [telus/tds-core#893]
## Description

Originally reported by Apurv and @ah-arch.

When using the TDS Input component, my design calls for more hint text than what is able to comfortably fit within a small parent container. 

## Recommendation

Provide a means for designers to optionally provide hint text below the Input's label, rather than inline.

TDS recommendation: add a new `description` prop that adds text below the label, and keep `hint` available for brief descriptive messages that display inline with the label. `description` would have a similar behaviour to core-radio's `description` prop.

Inline text (`hint`) and descriptive text (`description`) must not be used together. Only one or neither can be present in a given time.

## Designs

This has been approved by design leads.

![image](https://user-images.githubusercontent.com/12798751/51206798-fd1b3f00-18d6-11e9-8deb-41feb72679cf.png)

## Meta

- TDS component: @tds/core-input@1.0.13
- Willing to develop solution: Yes
- Has workaround: No 
- High impact: Yes 
- Designer/Developer Pair: Apurv Ray, @ah-arch

**Comment by ah-arch** [comment]
Can a user supply both a `description` and a `helper` or `hint`?
Where would the `helper` go - below or above the `description`?

**Comment by theetrain** [comment]
> Can a user supply both a description and a helper or hint?

We'll tackle this design problem in (issue number)

> Where would the helper go - below or above the description?

Here's an alternative prop interface suggested by @jraff: `hintPosition: PropTypes.oneOf(['inline', 'below'])` in which you can manually set the hint position to be inline or below, relative to the input label.

As a future enhancement, we could even add a 'responsive' prop value, which could work like this:

```jsx
<Input 
  label="First name"
  value="Enrico"
  hint="Your first name"
  hintPosition="responsive" />
// change hint text position automatically based on parent width and character count

<Input
  label="First name"
  value="Enrico"
  hint="Your first name"
  hintPosition={{xs: 'below', md: 'inline'}} />
// change hint text position based on breakpoint using responsiveProps
// See @tds/util-prop-types https://tds.telus.com/components/index.html#proptypes
```

> And this prop should also apply to TextArea and Select?

Correct, since core-input is a higher order component of FormField, as are other TDS form components, these patterns should apply to all other form components as well.

**Comment by ah-arch** [comment]
Is this wording okay for **Usage Criteria**?
> The `hint` should not be more than 85 characters when positioned below the label

**Comment by theetrain** [comment]
This has been resolved in [@tds/core-input@2.1.0](https://github.com/telus/tds-core/releases/tag/%40tds%2Fcore-input%402.1.0)
